src/rpc/util.cpp

```diff
@@ -40,21 +40,15 @@ void RPCTypeCheck(const UniValue& params,
             break;
 
         const UniValue& v = params[i];
-        if (!(fAllowNull && v.isNull())) {
-            RPCTypeCheckArgument(v, t);
+        if (fAllowNull && v.isNull()) continue;
+        if (!t.typeAny && v.type() != t.type) {
+            throw JSONRPCError(RPC_TYPE_ERROR,
+                               strprintf("JSON value of type %s is not of expected type %s", uvTypeName(v.type()), uvTypeName(t.type)));
         }
         i++;
     }
 }
 
-void RPCTypeCheckArgument(const UniValue& value, const UniValueType& typeExpected)
```

Comment:
not really a fan of removing this, given that this is needed for https://github.com/bitcoin/bitcoin/pull/26039

Reply:
Left you a comment there, https://github.com/bitcoin/bitcoin/pull/26039#discussion_r971027963.
I think that we shouldn't continue having the type error string defined in two different places.

Reply:
I am removing `RPCTypeCheck` and you are adding the type error string to `RPCTypeCheck`. So I don't see how the changes here require removing it or help in any way to reducing it to one place.

Reply:
Also, unrelated: `RPCTypeCheckObj` has a third and different type error string?

Reply:
> I am removing `RPCTypeCheck` and you are adding the type error string to `RPCTypeCheck`. So I don't see how the changes here require removing it or help in any way to reducing it to one place.

Yeah, it's not really needed here. I was pointing out that, as you are removing `RPCTypeCheck`, and I'm removing `RPCTypeCheckArgument` here, we could also reduce the error string duplication there as well.

But.. I could also implement the changes here too..
So yeah, probably better to leave it up for another PR and done.

> Also, unrelated: RPCTypeCheckObj has a third and different type error string?

Yeah, I mentioned it in one of the first comments of this PR https://github.com/bitcoin/bitcoin/pull/25737#issuecomment-1200425927 and then forgot about it.
Could fix it on the follow-up `Univalue::checkType` unification PR.